plugins/source/k8s/client/helpers.go

```diff
@@ -52,34 +52,81 @@ func ResolveContext(_ context.Context, meta schema.ClientMeta, r *schema.Resourc
 	return r.Set(c.Name, client.Context)
 }
 
-// In k8s, IP Addresses may sometimes be empty-strings - but postgresql doesn't like that.
-// So, the resolver for ip-addresses should recognize that case and not set null instead.
+// In k8s, IP Addresses may sometimes be empty-strings or `None` - but postgresql doesn't like that.
+// So, the resolver for ip-addresses should recognize that case and set null instead.
 func StringToInetPathResolver(path string) schema.ColumnResolver {
 	return func(_ context.Context, meta schema.ClientMeta, r *schema.Resource, c schema.Column) error {
 		value := funk.Get(r.Item, path, funk.WithAllowZero())
 
 		stringValue, ok := value.(string)
-		if ok && stringValue != "" {
+		if ok && stringValue != "" && stringValue != "None" {
 			return r.Set(c.Name, value)
 		}
+
 		return r.Set(c.Name, nil)
 	}
 }
 
-// In k8s, IP Addresses may sometimes be empty-strings - but postgresql doesn't like that.
-// So, the resolver for ip-addresses should recognize that case and not set null instead.
+// In k8s, IP Addresses may sometimes be empty-strings or `None` - but postgresql doesn't like that.
+// So, the resolver for ip-addresses should recognize that case and set null instead.
+func StringToInetArrayPathResolver(path string) schema.ColumnResolver {
+	return func(_ context.Context, meta schema.ClientMeta, r *schema.Resource, c schema.Column) error {
+		value := funk.Get(r.Item, path, funk.WithAllowZero())
+
+		stringArrayValue, ok := value.([]string)
+		if !ok {
+			return r.Set(c.Name, nil)
+		}
+
+		filteredStringArray := []string{}
```

Comment:
Yeah, this solution doesn't mirror the API exactly. However, I believe this is "good enough".

- Semantically, `k8s` state that there may be a difference between `None` and `""`. https://github.com/kubernetes/api/blob/59fcd23597fd090dba6b7e903eb0a8c9e8efb0a6/core/v1/types.go#L4529-L4544

- for the "non-array" version, we can't distinguish between `None` and `""` - we only have one "null" to work with. Thus, not sure it's super-important to distinguish them in the array-version (if we don't do it everywhere).

Reply:
@hermanschaaf No solution is perfect here I think (turning the `inets` into strings in order to be 100% compliant to the API is also breaking and annoying).

I could go either way honestly, but prefer the silent/non-breaking one..

Reply:
I guess what I'm saying is that we seem to be dropping both the `""` and `"None"` IPs in the array version, rather than storing them as `nil` values in the array, which I believe our type system now allows. I'm not too worried about distinguishing between `""` and `"None"`